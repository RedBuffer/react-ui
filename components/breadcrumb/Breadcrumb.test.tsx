import * as React from 'react';
import { shallow, mount } from 'enzyme';

import { Breadcrumb, BreadcrumbItem } from './index';

describe('Breadcrumb', () => {
  it('should render correctly', () => {
    const component = shallow(<Breadcrumb />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with BreadcrumbItem', () => {
    const component = mount(
      <Breadcrumb>
        <BreadcrumbItem>
          <a href="#">crumb 1</a>
        </BreadcrumbItem>
      </Breadcrumb>,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with BreadcrumbItem and string separator', () => {
    const component = mount(
      <Breadcrumb separator={'///'}>
        <BreadcrumbItem>
          <a href="#">crumb 1</a>
        </BreadcrumbItem>
      </Breadcrumb>,
    );

    expect(component).toMatchSnapshot();
  });
});
