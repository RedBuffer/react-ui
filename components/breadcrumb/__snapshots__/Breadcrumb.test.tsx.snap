// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render correctly 1`] = `
<div
  className="m-0 p-2 box-border font-medium react-ui-breadcrumb"
>
  <span
    className="text-primary"
  />
</div>
`;

exports[`Breadcrumb should render correctly with BreadcrumbItem 1`] = `
<Component>
  <div
    className="m-0 p-2 box-border font-medium react-ui-breadcrumb"
  >
    <span
      className="text-primary"
    >
      <Component>
        <span
          className="react-ui-breadcrumb-item"
        >
          <a
            href="#"
          >
            crumb 1
          </a>
        </span>
      </Component>
    </span>
  </div>
</Component>
`;

exports[`Breadcrumb should render correctly with BreadcrumbItem and string separator 1`] = `
<Component
  separator="///"
>
  <div
    className="m-0 p-2 box-border font-medium react-ui-breadcrumb"
  >
    <span
      className="text-primary"
    >
      <Component>
        <span
          className="react-ui-breadcrumb-item"
        >
          <a
            href="#"
          >
            crumb 1
          </a>
        </span>
      </Component>
    </span>
  </div>
</Component>
`;
